src/main/java/org/roaringbitmap/RoaringIterator.java

```diff
@@ -0,0 +1,138 @@
+package org.roaringbitmap;
+
+import java.util.NoSuchElementException;
+
+class RoaringIterator implements IntIterator {
+  final private RoaringBitmap parent;
+  private ArrayContainer ac;
+  private RunContainer rc;
+  private int containerIndex;
+  private short typecode = TypeCode.NONE;
+  private int hb;
+  private int le = 0;
+  private int maxlength;
+  private int base;
+  private long w;
+  private int pos;
+  private long[] bitmap;
+
+  RoaringIterator(RoaringBitmap parent) {
+    this.parent = parent;
+    this.hb = 0;
+    this.containerIndex = 0;
+    nextContainer();
+  }
+
+  public boolean hasNext() {
+    return containerIndex <= parent.highLowContainer.size;
+  }
+
+  public int next() {
+    int answer;
+    switch (typecode) {
+      case TypeCode.BITSET_CONTAINER:
+        long t = w & -w;
```

Comment:
First it was just port from C, but it seems that by this approach I'm able to call `nextContainer();` without additionally checking already checked conditions. Using polymorphism is what Flyweight implementation already does and it seems that this PR is on the good way to beat performance. Anyway, I suggest to keep this review and try to answer it once again at the end.